gnat2goto/driver/tree_walk.adb

```diff
@@ -720,31 +837,37 @@ package body Tree_Walk is
          Pos_Number := Pos_Number + 1;
       end loop;
 
-      --  We now have an array literal of some sort (Array_Expr).
-      --  Now encase it in an array structure,
-      --  and allocate memory to hold the literal:
-      Append_Struct_Member (Result_Struct, Low_Expr);
-      Append_Struct_Member (Result_Struct, High_Expr);
-      declare
-         Dup : constant Irep :=
-           Get_Array_Dup_Function (Element_Type_Ent, Index_Type_Node);
-         Call_Args : constant Irep := New_Irep (I_Argument_List);
-         Call_Expr : constant Irep :=
-           Make_Side_Effect_Expr_Function_Call (I_Function => Dup,
-                                                Arguments => Call_Args,
-                                                Source_Location => Sloc (N));
-      begin
-         Append_Argument (Call_Args, Typecast_Expr);
-         Append_Argument (Call_Args, Len_Expr);
-         Append_Struct_Member (Result_Struct, Call_Expr);
-      end;
-
-      return Make_Let_Expr (Symbol => Literal_Temp,
```

Comment:
The downside of the rewrite:
Not so happy that we never resolved the original issue why the "let_expr" and allocated array parameter passing failed. 
I think we are now missing a tool in the toolbox - would be good to raise an issue for this.